packages/eslint-plugin-react-hooks/src/RulesOfHooks.js

```diff
@@ -295,7 +305,7 @@ export default {
           if (pathList.has(segment.id)) {
             const pathArray = Array.from(pathList);
             const cyclicSegments = pathArray.slice(
-              pathArray.indexOf(segment.id) - 1,
+              pathArray.indexOf(segment.id) + 1,
```

Comment:
Reverts the change from #28714 as it was introducing false positives (see https://github.com/facebook/react/issues/31687).